sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Artifact.java

```diff
@@ -59,19 +60,18 @@ public final Release addRelease(Release release) {
     return release;
   }
 
-  public final Release addRelease(Version version) {
-    return addRelease(new Release(this, version));
-  }
-
-  public final Release addRelease(String version) {
-    return addRelease(new Release(this, version));
+  /**
+   * @throws java.util.NoSuchElementException if release could not be found
+   */
+  public final Release getRelease(Version version) {
+    return getRelease(version, null);
   }
 
   /**
    * @throws java.util.NoSuchElementException if release could not be found
    */
-  public final Release getRelease(Version version) {
-    for (Release release : getAllReleases()) {
+  public final Release getRelease(Version version, @Nullable Product product) {
```

Comment:
Is a Null product something possible or it is supposed to thrown every time the NoSuchElementException?

Reply:
Discussed offline.